File: iconics-java/src/main/java/com/open/iconics/internal/CompoundIconicsDrawables.java
package com.open.iconics.internal;

import androidx.annotation.Nullable;
import androidx.annotation.RestrictTo;

import com.open.iconics.IconicsDrawable;

import static androidx.annotation.RestrictTo.Scope.LIBRARY_GROUP;

@RestrictTo(LIBRARY_GROUP)
public interface CompoundIconicsDrawables {

    @Nullable
    IconicsDrawable getIconicsDrawableStart();

    @Nullable
    IconicsDrawable getIconicsDrawableTop();

    @Nullable
    IconicsDrawable getIconicsDrawableEnd();

    @Nullable
    IconicsDrawable getIconicsDrawableBottom();

    void setDrawableStart(@Nullable IconicsDrawable drawable);

    void setDrawableTop(@Nullable IconicsDrawable drawable);

    void setDrawableEnd(@Nullable IconicsDrawable drawable);

    void setDrawableBottom(@Nullable IconicsDrawable drawable);

    void setDrawableForAll(@Nullable IconicsDrawable drawable);
}

File: iconics-kotlin/src/main/java/com/open/iconics/internal/CompoundIconicsDrawables.kt
package com.open.iconics.internal

import com.open.iconics.IconicsDrawable

internal interface CompoundIconicsDrawables {

    var iconicsDrawableStart: IconicsDrawable?

    var iconicsDrawableTop: IconicsDrawable?

    var iconicsDrawableEnd: IconicsDrawable?

    var iconicsDrawableBottom: IconicsDrawable?

    fun setDrawableForAll(drawable: IconicsDrawable?)
}
